Make config.fx_graph_remote_cache be three-value switch

oulgen opened the pull request:
Summary:
We want to allow for three configurations
False: Force off
True: Force on
None: OFF for OSS and JK config for internal

Test Plan: CI

Differential Revision: D58535897




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

masnesral commented on test/inductor/test_codecache.py:
```diff
@@ -109,6 +109,7 @@ def reset(self):
 
     @requires_triton()
     @config.patch({"fx_graph_cache": True})
+    @config.patch({"fx_graph_remote_cache": False})
```
FYI I think this config.patch thing takes dictionaries, so you could combine with the line above if you happen to like that better; I don't care-- just FYI

oulgen committed:
Make config.fx_graph_remote_cache be three-value switch (#128628)

Summary:

We want to allow for three configurations
False: Force off
True: Force on
None: OFF for OSS and JK config for internal

Test Plan: CI

Reviewed By: eellison, masnesral

Differential Revision: D58535897
